src/app.module.ts

```diff
@@ -1,10 +1,56 @@
-import { Module } from '@nestjs/common';
+import { Logger, Module } from '@nestjs/common';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
+import { UsersModule } from './users/users.module';
+import { MongooseModule } from '@nestjs/mongoose';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { AuthModule } from './auth/auth.module';
+import { AccessControlModule } from './access-control/access-control.module';
+import { BranchModule } from './branch/branch.module';
+import { AttendanceModule } from './attendance/attendance.module';
+import { UserRepository } from './users/users.repository';
+import { ROLE } from './constants';
 
 @Module({
-    imports: [],
+    imports: [
+        ConfigModule.forRoot({
+            isGlobal: true,
+        }),
+        MongooseModule.forRootAsync({
+            imports: [ConfigModule],
+            useFactory: async (configService: ConfigService) => ({
+                uri: configService.get<string>('MONGODB_URI'),
+            }),
+            inject: [ConfigService],
+        }),
+        UsersModule,
+        AuthModule,
+        AccessControlModule,
+        BranchModule,
+        AttendanceModule,
+    ],
     controllers: [AppController],
     providers: [AppService],
 })
-export class AppModule {}
+export class AppModule {
+    private readonly logger = new Logger(AppModule.name);
```

Comment:
@kevit-shrut-sureja use nest lifecycle hooks

Reply:
solved this in the commit 487feab39c87f6e5f0d6b03e58cbcca982098f4a